### PR TITLE
feat: Upload all artifacts of release builds to s3 - not just dylibs

### PR DESCRIPTION
**Pull request by anton-chain**
# Pull Request

Closes: PRO-1607

## Checklist

Please conduct a thorough self-review before opening the PR.

- [ ] I am confident that the code works.
- [ ] I have written sufficient tests.
- [ ] I have written and tested required migrations.
- [ ] I have updated documentation where appropriate.

## Summary

*Please include a succinct description of the purpose and content of the PR. What problem does it solve, and how? Link issues, discussions, other PRs, and anything else that will help the reviewer.*

#### Non-Breaking changes

If this PR includes non-breaking changes, select the `non-breaking` label. On merge, CI will automatically cherry-pick the commit to a PR against the release branch.

### CODE REVIEW

File: .github/workflows/_61_upload_engine_shared_libs.yml [hunk 1]
```diff
@@ -46,4 +35,4 @@ jobs:
 
       - name: Upload engine shared libraries S3 🚀
         run: |
-          aws s3 cp /tmp/chainflip-engine-shared-libs/ s3://repo.chainflip.io/lib/${{ steps.version.outputs.url_friendly_version }}/ --recursive
+          aws s3 cp ./artifacts/ s3://repo.chainflip.io/lib/${{ env.url_friendly_version }}/ --recursive
```
[the remark this answers is not in the document]
Comment (anton-chain) [reply .github/workflows/_61_upload_engine_shared_libs.yml:38]: validation was failing somehow, also thought `env` is a bit more unified without connection to particular step ID.

File: .github/workflows/_61_upload_engine_shared_libs.yml [hunk 2]
```diff
@@ -7,36 +7,25 @@ permissions:
...
+          pattern: chainflip-backend-bin*
+          merge-multiple: false
 
       - name: Set Version 🔦
         id: version
-        shell: bash
         run: |
-          chmod +x chainflip-backend-bin/chainflip-node
-          VERSION=$(./ci/scripts/extract_version.sh "$(./chainflip-backend-bin/chainflip-node --version)" full)
+          chmod +x artifacts/chainflip-backend-bin/chainflip-node
```
[the remark this answers is not in the document]
Comment (anton-chain) [reply .github/workflows/_61_upload_engine_shared_libs.yml:25]: though this step is actually needed to execute binary in a pipeline and get some extra output.

File: .github/workflows/_61_upload_engine_shared_libs.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (anton-chain) [reply .github/workflows/_61_upload_engine_shared_libs.yml:38]: It look like this ATM:
```
$ aws s3 ls s3://repo.chainflip.io/lib/
                           PRE 1_4_6/
                           PRE 1_5_0/
                           PRE 1_5_1/
                           PRE 1_6_0/
                           PRE 1_6_1/
```
shall we construct it to move `${{ env.url_friendly_version }}` to the top level? we have other folder there like:
```
$ aws s3 ls s3://repo.chainflip.io/
                           PRE chainflip/
                           PRE chainspecs/
                           PRE engine/
                           PRE keys/
                           PRE kitkat/
                           PRE legacy/
                           PRE lib/
                           PRE partnernet/
                           PRE perseverance/
                           PRE releases/
                           PRE test/
                           PRE tools/
2024-03-25 14:25:00   93596763 chainflip-backend-bin-ubuntu-22.04.zip
2022-11-08 11:50:28        254 index.html
2024-08-15 21:15:00 813538867200 nitro-pruned.tar
```

File: .github/workflows/_61_upload_engine_shared_libs.yml
Comment (kylezs) [inline .github/workflows/_61_upload_engine_shared_libs.yml:1]: We should rename this file now too

Comment (kylezs) [reply .github/workflows/_61_upload_engine_shared_libs.yml:1]: upload_versioned_build_artifacts.yml ?

**Pull request merged**